tkg/client/upgrade_cluster_clusterclass.go

```diff
@@ -21,7 +22,10 @@ func (c *TkgClient) DoClassyClusterUpgrade(regionalClusterClient clusterclient.C
 	log.Infof("Upgrading kubernetes cluster to `%v` version, tkr version: `%s`", kubernetesVersion, tkrVersion)
 	patchJSONString := fmt.Sprintf(`{"spec": {"topology": {"version": "%v"}}}`, tkrVersion)
 
-	err := regionalClusterClient.PatchClusterObject(options.ClusterName, options.Namespace, patchJSONString)
+	// Timeout set to 15 minutes because the continuousTKRDiscoverFreq for tkr-source-controller's fetcher is 10 minutes.
```

Comment:
This PR looks good to me. I just cannot understand how timeout is related to the tkr fetcher's frequency.

Reply:
https://github.com/vmware-tanzu/tanzu-framework/blob/decac8230fd187b1e9ac19d786f32358b37298e2/tkr/controller/tkr-source/main.go#L71

In tkr-source-controller the fetcher tried to sync the tkrs from registry every 10 mins. This part was meant to cover at least 1 frequency period to let fetcher do a fetch action(and download) if the desired tkrs are not found.

Reply:
@hongtaiwang @12345lcr  why the timeout is related to the fetcher frequency here?

Reply:
By patching `TKGVERSION` annotation before tkr-source-controller creation, we do not need to extend this time out, right?

Reply:
Per discussion, in worst case, this timeout change can still make the tkr image download safer.